core/src/main/clojure/xtdb/buffer_pool.clj

```diff
@@ -329,18 +346,20 @@
 
         :else
         (let [buffer-cache-path (.resolve local-disk-cache k)]
-          (-> (if (util/path-exists buffer-cache-path)
-                ;; todo could this not race with eviction? e.g exists for this cond, but is evicted before we can map the file into the cache?
-                (CompletableFuture/completedFuture buffer-cache-path)
+          (-> (if (.getIfPresent local-disk-cache-evictor buffer-cache-path)
```

Comment:
Regarding race conditions and missing files: I suspect you run into the race where `getIfPresent` returns true (thread 1), some other thread removes the file on disc (thread 2) and then you are trying to mmap it again. If you want to make this thread safe you can look at 
```clj
(.. local-disk-cache asMap compute/computeIfPresent (reify BiFunction ....))
```
We did something similar in my PR: https://github.com/FiV0/xtdb/blob/c21398a6dd51f009845de4c5625e7a2c631c8b29/core/src/main/clojure/xtdb/buffer_pool.clj#L47

Reply:
Yeah, this was absolutely a problem - saw this a lot more frequently locally when fixing up the tests. Changed this now, shall mark as resolved - thanks @FiV0 :)